components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java

```diff
@@ -4868,7 +4868,15 @@ public static NativeObject jsFunction_getDocument(Context cx, Scriptable thisObj
         }
         String resource = (String) args[1];
         String tenantDomain = (String) args[0];
-        Map<String, Object> docResourceMap = APIUtil.getDocument(username, resource, tenantDomain);
+        boolean isTenantFlowStarted = false;
+        try {
+            if(tenantDomain != null && !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+                isTenantFlowStarted = true;
```

Comment:
since the tenant flow hasn't started yet. Better move this to after PrivilegedCarbonContext.startTenantFlow();